### PR TITLE
Openssl_devel  - headers to allow building against system openssl

### PR DESCRIPTION
**Pull request by minektur**
Installs JUST the headers to match the chromeos supplied libraries so that you can build things 
that link with openssl. Needs to be kept version-synced with chromeos releases
Could detect current version and grab one of many different packages, compare to saved
hashes and support multiple versions  maybe - call it a TODO

update the two 'from-source' python pacakges and libevent to depend on openssl_devel

### CODE REVIEW

**Comment by minektur** [comment]
In a different pull request I just disabled the openssl stuff in libevent so that I could build tmux.

Doing it this way is arguably better.

**Comment by lyxell** [comment]
Could incorrectly versioned headers break anything?

A temporary security check could be to ensure that running `/usr/bin/openssl version` reports a compatible version prior to installing, not very robust, but better than nothing.

**Comment by minektur** [comment]
Between minor versions no...  The only time it would make a difference is if the api's change or if the datastructures change, and then building against openssl may or may not work.  Existing binaries would be fine - just new builds would possibly break in subtle ways, or dramatic ways - no telling without knowing the future (e.g. what changes are coming).  There is a reasonable chance that the builds would just fail also.

So you have libraries from one version of openssl and .h files from another - there would be some mismatch, but it is somewhat unpredictable.

**Comment by lyxell** [comment]
Alright, everything looks good to me, this approach is way better than overriding the system OpenSSL executable which is what we're currently doing.

**Comment by skycocker** [comment]
@minektur could you merge master into this PR?

**Comment by minektur** [comment]
Yes - I'll whip up a new version of this PR

File: packages/openssl_devel.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  def self.install
+    out = "#{CREW_DEST_DIR}/usr/local/include/openssl"
+    system "mkdir -p #{out}"
+
+    `ls ./include/openssl`.split(' ').each do |header|
+      system "cp", "./include/openssl/#{header}", out
+    end
+
+    #system "cp", "./include/openssl/*", "#{out}"
```
Comment (skycocker) [inline packages/openssl_devel.rb]: could we delete this line or it might be needed someday?

**Comment by thedamian** [comment]
Merged

**Pull request merged**